File: Contain_Duplicate/contain_dup.dart
class Solution {
  bool containsNearbyDuplicate(List<int> nums, int k) {
    final indexTable = {};
    for (var i = 0; i < nums.length; i++) {
      if (indexTable.containsKey(nums[i])) {
        if (i - indexTable[nums[i]] <= k) {
          return true;
        }
      }
      indexTable[nums[i]] = i;
    }

    return false;
  }
}

File: Contain_Duplicate/contain_dup.go
package containduplicate

func containsNearbyDuplicate(nums []int, k int) bool {
	indexTable := map[int]int{}

	for index, num := range nums {
		tableIndex, exist := indexTable[num]

		if exist {
			if index-tableIndex <= k {
				return true
			}
		}
		indexTable[num] = index
	}

	return false
}
